projects/storefrontlib/cms-components/product/product-list/container/product-list-component.service.ts

```diff
@@ -84,10 +85,26 @@ export class ProductListComponentService {
           state.params,
           state.queryParams
         );
-        this.search(criteria);
+        this.searchIfQueryHasChanged(criteria);
       })
     );
 
+  /**
+   * Search only if the previous search query does NOT match the new one.
+   * This prevents repeating product search calls for queries that already have loaded data.
+   */
+  protected searchIfQueryHasChanged(criteria: SearchCriteria) {
+    this.productSearchService
+      .getResults()
+      .pipe(take(1))
+      .subscribe((results) => {
+        const loadedQuery = results?.currentQuery?.query?.value;
+        if (loadedQuery !== criteria.query) {
+          this.search(criteria);
+        }
+      });
+  }
```

Comment:
Nice idea. The solution feels to me like very close to the roots of the issue, not fixing the symptoms. We check if we really need to reload the data, instead of checking `TransferState` issues (which is only a derivative of the root problem). I like this solution doesn't leak information about using binding TransferState to ngrx state (which is responsibility of an other layer of the application).